### PR TITLE
The BIG KITCHEN UPDATE: Part One, the machines and food

### PR DESCRIPTION
**Pull request by SigDoesCode**
## About The Pull Request
This is part one of two of the big kitchen and cooking update. Currently, this adds the machines,  the tools for the machines, and the recipes made in the machines. It also adds the food associcated with the recipes. The microwaves are still usable, but not for the new food, and there will be no microwaves on the new kitchen map that I intend to implement after this addition.
The new machines:

The way that cooking works now is as follows: 
1. Pick a recipe. Once you've decided what you want to cook, you need to...
2. Pick the appropriate machine. Various recipes have different machine requirements. If you want to make a specific recipe, you need to pick the machine appropriate to the recipe. Alternately, you can decide to use a machine for specific kind of combination cooking where you select a recipe and add ingredients to make, for example, an egg and cheese loaf. 
image

3. Once you've picked your machine, you need to turn it on. 
gif

4. Put your ingredients into the machine. This will apply the machine's effect to it, grilling on the grill, baking in the oven, etc. IT comes with a name change and a sprite overlay. 
4.5. If instead you want to make a recipe, you need to remove a cooking container from the machine through a handy radial menu. Insert the ingredients into the tray and put it back in.
image
Above: the grill with meat
image
Above: the grill with a finished grilled raw meat steak. To properly cook it, you need to add more ingredients.

5. Now, wait! Once you've put your ingredients in, the machine will start cooking. The more ingredients you have the longer it takes to cook, so for batch baking or large recipes, go do something else while it cooks. You can examine the machines while they cook to check the status of each ingredient. 
6. Listen for the ding. Once a machine is done, it will ding. If you leave the food in for too long, though, the food will burn! 
image
Above: The grill warning the player that it is done. Note: There is a ping sound associated, but I can't put it in here.
image
Above: I looked away from the oven for a moment while I was doing the PR and something burned. Also, this shows off the examine menu. 
image
Above: The examine menu while making food. Notice the blue text showing you how the food inside is doing.
Also, the whole thing is modular so it should be totally okay to do even after the massive rebase PR thing is okay to go. I'm mostly asking for this to be merged so that I can work on the implementation of the new map stuff to help test things. 

Features of cooking machines:
- control click to turn them on
- Alt click to open the oven door
- Examine them to check on your food
- Food will cook without you having to look at it
- Multiple slots for cooking containers
- Specific recipe selection on right click
- default cooking for adding prefixes and overlays
- Recipe cooking if all the ingredients for a recipe are added to a single container
- The ability to make multiple things at a time! 

Features of cooking containers:
- Sprites different based on what kind of container they are.
- Different cooking containers can't fit in inappropriate machines
- You can put food into the containers directly, but clicking on the machine will put it in the highest-number empty container
- Alt+click the container to entirely remove the food contents
- They're also reagent containers so you can pour things into and out of them. 

Kinds of machines added:
Grill: This is the grill. It grills food. Things made on this machine are grilled. Typically, this includes meats. It can hold one cooking container at a time, the grill grate, but this container can hold three things at once. 
image

Oven: This is an oven, you make cookies and things in it. It adds the baked modifier to food. It's used to make things like bread, pies, and pizzas.
The oven has a massive FIVE cooking containers at once, meaning you can make multiple things with it at the same time.
image
However... 
image
Once you turn it on, you have to turn it off to open it again, so be careful. 

Plancha: This is a plancha, a large flattop cooking surface for making things. It contains three plancha dividers to make three things at once. You can use this for things like rice, eggs, and omelettes. The plancha sears things.
image

Fryer: This is a deep fat fryer. It contains two baskets that you can use to make things like fries, fish sticks, and other fried foods. 
image
The overlay also changes when you remove the baskets. Technically, it runs on corn oil, but this is a bit sticky at the moment and requires more testing. The kitchen is going to start with a large amount of oil and be able to make more.  The fryer currently fries things. 

Stove: This is the stove, where you can put things in skillets, pots, and saucepants. Things put on the stove are pan fried. 
image
It has four slots for cooking containers and the overlay updates depending on what's on it! It's the most 

Mixer: This is a massive stand mixer. It's for making doughs and breads and things, but currently it has no associated recipes as far as I can tell. The intent is to use it to make large amounts of dough at once for cooking. Later we can add more things to it. 
image

Also, it spin. 
gif

Combination Cooking: 
This is a system from Aurora where you can pick a base food and add things to it in a cooking process. For example, picking the base of waffle and adding and meat and a chocolate bar will create this monstrosity: 
image
As of this update I'm still trying to get it to mix the taste tags of the ingredients. 

Technically, it combines the reagents and filling colors of the ingredients and applies them to a base sprite in food_custom.dmi and then spits it out. It also combines the names. 
image

## Why It's Good For The Game
Kitchen gameplay should now be way more interesting and customizeable. 

## Changelog
```changelog
add: Added nearly 200 recipes to the kitchen
add: Added more ways of making food
del: Removed microwaves from the kitchen
```

### CODE REVIEW

File: code/modules/food/recipes_microwave.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@ I said no!
 	)
 	result = /obj/item/reagent_containers/food/snacks/telebacon
 */
-
```
Comment (WeNeedMorePhoron) [inline code/modules/food/recipes_microwave.dm:21]: This file as a whole need an Occulus Edit start / Occulus Edit End just so it don't get overridden by accident

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -25,7 +25,6 @@
 
 	var/sanity_gain = 0.2 //per nutriment
 	var/junk_food = FALSE //if TRUE, sanity gain per nutriment will be zero
-	var/cooked = FALSE
```
Comment (WeNeedMorePhoron) [inline code/modules/reagents/reagent_containers/food/snacks.dm:28]: Need this to be marked as Occulus Edit since it still exists upstream

Comment (WeNeedMorePhoron) [reply code/modules/reagents/reagent_containers/food/snacks.dm:28]: Sanity gain per bite code seems to be still in. Is this intentional?

**Review by WeNeedMorePhoron — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 2 above)
Comment (SigDoesCode) [reply code/modules/reagents/reagent_containers/food/snacks.dm:28]: That's intentional, and I thought I didn't have any noncore files. That's something I can fix.

**Comment by SigDoesCode** [comment]
I'm not 100% sure what to do with these merge conflicts

File: code/modules/food/recipes_microwave.dm (hunk 1 above)
Comment (SigDoesCode) [reply code/modules/food/recipes_microwave.dm]: Replaced this, the file is reenabled and the entire change is modular.

**Review by WeNeedMorePhoron — COMMENTED** [review]
Will review and test locally when it compiles without error

**Comment by SigDoesCode** [comment]
> Will review and test locally when it compiles without error

Compiles without error? What errors are you getting? Also, I'm unsure why the linters are failing.

File: zzzz_modular_occulus/code/__HELPERS/kitchen.dm [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/datum/reagent/proc/set_thermal_energy(amount, var/datum/reagents/holder, safety = FALSE)
+	var/delta = amount/get_thermal_fraction(holder) - holder.chem_temp
+	if(!round(delta, 1))
+		return
+	holder.add_thermal_energy(delta, safety)	//Handles on_heat_change
+
+/datum/reagent/proc/add_thermal_energy(amount, var/datum/reagents/holder, safety = FALSE)
+	holder.add_thermal_energy(amount, safety)
+
+datum/reagent/proc/set_temperature(temperature, var/datum/reagents/holder, safety = FALSE)
```
Comment (WeNeedMorePhoron) [inline zzzz_modular_occulus/code/__HELPERS/kitchen.dm:18]: Missing /

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 2 above)
Comment (WeNeedMorePhoron) [reply code/modules/reagents/reagent_containers/food/snacks.dm:28]: Deletion of this line need to be a comment and marked as Occulus Edit

**Review by WeNeedMorePhoron — CHANGES_REQUESTED** [review]
Just nitpick - Tested locally with grilled meat. Seems like most of the new food don't have any base sanity gain, might want to look into it. 

Grill and any other items that have items in it removed by Alt-Click should be description saying so.

Of note is that Microwave now seems to follow your recipes - microwaving salt + pepper + meat = burned mess, while microwaving a plain cutlet resulted in chicken katsu. 

Other bugs detected locally: 
Microwaving Human Meat + Bun created an invisible burger called "The Snack" . Plain burger seems undefined.

**Comment by SigDoesCode** [comment]
I'll look at this cooked thing, I don't know if I have it somewhere else or not but I know I have the var in. I don't know why there's a change in that file.

**Comment by SigDoesCode** [comment]
There, I fixed the issues and added a thing to the descriptions.

**Comment by WeNeedMorePhoron** [comment]
Looks like there's compilation errors with ethanol instead of alcohol typepath. This is due to alcohol rework that went in a while back.

**Review by WeNeedMorePhoron — APPROVED** [review]
we balling

**Pull request merged**